src/backend/executor/bridge.ts

```diff
@@ -107,21 +108,36 @@ export class ExecutorBridge implements Disposable {
         return undefined;
     }
 
-    public getAnalyzeCmdLine(...files: Uri[]): string | undefined {
+    public getAnalyzeCmdArgs(...files: Uri[]): string[] | undefined {
         if (!workspace.workspaceFolders?.length) {
             return undefined;
         }
 
         // TODO: Refactor for less code repetition across functions
         const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
 
-        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
-            ?? 'CodeChecker';
         const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
             ?? path.join(workspaceFolder, '.codechecker');
-        const ccArguments = getConfigAndReplaceVariables('codechecker.executor', 'arguments') ?? '';
-        const ccThreads = workspace.getConfiguration('codechecker.executor').get<string>('threadCount');
 
+        const ccArgumentsSetting = workspace.getConfiguration('codechecker.executor').get<string>('arguments');
```

Comment:
By the looks of it, you will unconditionally `parse()` the user-specified string.
However, according to the _`Also accepts arguments in a string-array form.`_, it might not need any _shell-quote parsing_.

Reply:
Updated the tooltip below.